libs/langchain-ollama/src/chat_models.ts

```diff
@@ -0,0 +1,543 @@
+import {
+  AIMessage,
+  UsageMetadata,
+  type BaseMessage,
+} from "@langchain/core/messages";
+import {
+  BaseLanguageModelInput,
+  ToolDefinition,
+} from "@langchain/core/language_models/base";
+import { CallbackManagerForLLMRun } from "@langchain/core/callbacks/manager";
+import {
+  type BaseChatModelParams,
+  BaseChatModel,
+  LangSmithParams,
+  BaseChatModelCallOptions,
+} from "@langchain/core/language_models/chat_models";
+import { Ollama } from "ollama/browser";
+import { ChatGenerationChunk, ChatResult } from "@langchain/core/outputs";
+import { AIMessageChunk } from "@langchain/core/messages";
+import type {
+  ChatRequest as OllamaChatRequest,
+  ChatResponse as OllamaChatResponse,
+  Message as OllamaMessage,
+} from "ollama";
+import { StructuredToolInterface } from "@langchain/core/tools";
+import { Runnable, RunnableToolLike } from "@langchain/core/runnables";
+import { convertToOpenAITool } from "@langchain/core/utils/function_calling";
+import {
+  convertOllamaMessagesToLangChain,
+  convertToOllamaMessages,
+} from "./utils.js";
+import { concat } from "@langchain/core/utils/stream";
+
+export interface ChatOllamaCallOptions extends BaseChatModelCallOptions {
+  /**
+   * An array of strings to stop on.
+   */
+  stop?: string[];
+  tools?: (StructuredToolInterface | RunnableToolLike | ToolDefinition)[];
+}
+
+export interface PullModelOptions {
+  /**
+   * Whether or not to stream the download.
+   * @default true
+   */
+  stream?: boolean;
+  insecure?: boolean;
```

Comment:
What's this?

Reply:
`insecure ` is one of their fields. idk what it does but they offer it!